src/content/en/updates/2017/09/abortable-fetch.md

```diff
@@ -0,0 +1,306 @@
+
+      if (event.request.method == 'GET' && url.pathname == '/about/') {
+        // Modify the URL
+        url.searchParams.set('from-service-worker', 'true');
+        // Fetch, but pass the signal through
+        event.respondWith(
+          fetch(url, { signal: event.request.signal })
+        );
+      }
+    });
+
+Follow [the spec](https://github.com/w3c/ServiceWorker/pull/1178) to track this – I'll add links to
+browser tickets once it's ready for implementation.
+
+## The history
+
+Yeah… it took a long time for this relatively simple API to come together. Here's why:
+
+### API disagreement
+
+As you can see [the GitHub discussion is pretty long](https://github.com/whatwg/fetch/issues/27).
+There a lot of nuance in that thread (and some lack-of-nuance), but the key disagreement is one
+group wanted the `abort` method to exist on the object returned by `fetch()`, whereas the other
+wanted a separation between getting the response and effecting the response.
+
+These requirements are incompatible, so one group wasn't going to get what they wanted. If that's
+you, sorry! If it makes you feel better, I was also in that group. But seeing `AbortSignal` fit the
+requirements of other APIs makes it seem like the right choice. Also, allowing chained promises to
+become abortable would become very complicated, if not impossible.
+
+If you wanted to return an object that provides a response, but can also abort, you could create a
+simple wrapper:
+
+    function abortableFetch(request, opts) {
+      const controller = new AbortController();
+      const signal = controller.signal;
+      const fetchOpts = Object.assign({ signal }, opts);
+
+      return {
+        abort: () => controller.abort(),
+        ready: fetch(request, opts)
+      };
+    }
+
+### False starts in TC39
+
+There was an effort to make a cancelled action distinct from an error. This included a third promise
+state to signify "cancelled", and some new syntax to handle cancellation in both sync and async
+code:
+
+<p><span class="compare-worse">Not real code</span> — proposal was withdrawn</p>
+
+    try {
+      // Start spinner, then:
+      await someAction();
+    }
+    catch cancel(reason) {
+      // Maybe do nothing?
+    }
+    catch(err) {
+      // Show error message
+    }
+    finally {
+      // Stop spinner
+    }
+
+The most common thing to do when an action is cancelled, is nothing. The above proposal separated
+cancellation from errors so you didn't need to handle abort errors specifically. `catch cancel` let
+you hear about cancelled actions, but most of the time you wouldn't need to.
+
+This got to stage 1 in TC39, but consensus wasn't achieved, and [the proposal was
+withdrawn](https://github.com/tc39/proposal-cancelable-promises).
+
+Rather than go through TC39 again and face potentially months or even years of further delay, we
+tackled the problem in the DOM, resulting in `AbortController`, and rejecting promises with
+`AbortError`.
```

Comment:
I'd phrase this differently. For me at least the issue is that without syntax, which TC39 was unwilling to provide, it's simply a library feature, and thus best done as part of the web platform. Without language integration, it doesn't make sense to do it in TC39.

Not sure on exact phrasing, but I think emphasizing that aspect, rather than the months/years of delay, makes sense.